saleor/graphql/meta/types.py

```diff
@@ -115,9 +120,19 @@ def resolve_private_metafields(root: ModelWithMetadata, info, *, keys=None):
         return _filter_metadata(root.private_metadata, keys)
 
     @classmethod
-    def resolve_type(cls, instance: ModelWithMetadata, _info):
+    def resolve_type(cls, instance: ModelWithMetadata, info):
         if isinstance(instance, ChannelContext):
             # Return instance for types that use ChannelContext
             instance = instance.node
+        if isinstance(instance, Checkout):
+            from ..checkout.types import Checkout as CheckoutType
+
+            return CheckoutType.resolve_type(instance, info)
         item_type, _ = resolve_object_with_metadata_type(instance)
         return item_type
+
+
+def get_valid_metadata_instance(instance):
+    if isinstance(instance, Checkout):
+        instance = instance.metadata_storage
```

Comment:
Wont fail in case of missing metadata_storage relation?

Reply:
It seems that it would. Good catch!